vllm/compilation/sequence_parallelism.py

```diff
@@ -12,31 +12,148 @@
 from vllm.distributed.parallel_state import (
     get_tensor_model_parallel_world_size)
 from vllm.logger import init_logger
+from vllm.platforms import current_platform
 
 from .vllm_inductor_pass import VllmInductorPass
 
 logger = init_logger(__name__)
 
 
-class AllReduceRMSNormPattern:
+class _SequenceParallelPatternHelper:
+    """Base helper for sequence parallelism patterns."""
 
     def __init__(self, epsilon: float, dtype: torch.dtype, device: str):
         self.epsilon = epsilon
         self.dtype = dtype
         self.device = device
-
-
-class EmbeddingAllReduceRMSNormPattern(AllReduceRMSNormPattern):
+        self.tp_group = get_tp_group()
+        self.tp_size = get_tensor_model_parallel_world_size()
+
+    def _all_reduce(self, x: torch.Tensor) -> torch.Tensor:
+        return tensor_model_parallel_all_reduce(x)
+
+    def _reduce_scatter(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.ops.vllm.reduce_scatter.default(
+            x,
+            dim=0,
+            world_size=self.tp_size,
+            group_name=self.tp_group.unique_name)
+
+    def _all_gather(self, x: torch.Tensor) -> torch.Tensor:
+        return torch.ops.vllm.all_gather.default(
+            x,
+            dim=0,
+            world_size=self.tp_size,
+            group_name=self.tp_group.unique_name)
+
+
+class _RMSNormOpHelper(_SequenceParallelPatternHelper):
+    """Helper for RMSNorm operations in sequence parallelism patterns."""
+
+    def _functional_rmsnorm(self, result_buffer, input_tensor, weight_tensor):
+        return torch.ops.higher_order.auto_functionalized(
+            torch.ops._C.rms_norm.default,
+            result=result_buffer,
+            input=input_tensor,
+            weight=weight_tensor,
+            epsilon=self.epsilon)
+
+    def _functional_fused_add_rmsnorm(self, input_tensor, residual_tensor,
+                                      weight_tensor):
+        return torch.ops.higher_order.auto_functionalized(
+            torch.ops._C.fused_add_rms_norm.default,
+            input=input_tensor,
+            residual=residual_tensor,
+            weight=weight_tensor,
+            epsilon=self.epsilon)
+
+
+class _RMSNormQuantOpHelper(_SequenceParallelPatternHelper):
+    """Helper for RMSNorm + Quantization operations in sequence parallelism patterns."""  # noqa: E501
```

Comment:
I guess my main question with this PR is: why is the RMSNorm + quant fusion part of the sequence parallel pass? Should it be a standalone pass instead?